src/common/cvt.cpp

```diff
@@ -2037,9 +2313,50 @@ static void string_to_format_datetime_pattern_matcher(std::string_view pattern,
 	invalid_pattern_exception(pattern, cb);
 }
 
+// These rules are taken from ISO/IEC 9075-2:2023(E) 9.52 Datetime templates
+static void validate_format_flags(FormatPatterns formatFlags, Callbacks* cb)
+{
+	if (static_cast<bool>((formatFlags & (FormatPatterns::HH12 | FormatPatterns::AM | FormatPatterns::PM))))
+	{
+		// If CT contains <datetime template 24-hour>, then CT shall not contain <datetime template 12-hour> or <datetime template am/pm>.
+		if (static_cast<bool>(formatFlags & FormatPatterns::HH24))
+			cb->err(Arg::Gds(isc_incompatible_format_patterns) << Arg::Str("HH24") << Arg::Str("HH/HH12/A.M./P.M."));
+
+		// If CT contains <datetime template 12-hour>, then CT shall contain <datetime template am/pm> and shall not contain <datetime template 24-hour>.
+		// If CT contains <datetime template am/pm>, then CT shall contain <datetime template 12-hour> and shall not contain <datetime template 24-hour>.
+		if (static_cast<bool>(!(static_cast<bool>(formatFlags & FormatPatterns::HH12)
+			== (formatFlags & FormatPatterns::AM || formatFlags & FormatPatterns::PM))))
+		{
+			cb->err(Arg::Gds(isc_pattern_cant_be_used_without_other_pattern_and_vice_versa)
+				<< Arg::Str("HH/HH12") << Arg::Str("A.M./P.M."));
+		}
+	}
+
+	// If CT contains <datetime template second of day>, then CT shall not contain any of the following:
+	// <datetime template 12-hour>, <datetime template 24-hour>, <datetime template minute>, <datetime
+	// template second of minute>, or <datetime template am/pm>.
+	if (static_cast<bool>(formatFlags & FormatPatterns::SSSSS))
```

Comment:
Are these `static_cast<bool>` necessary?

Reply:
Yes. I recently read an interesting [discussion](https://chat.stackoverflow.com/rooms/9345/discussion-between-nicol-bolas-and-potatoswatter) about `enum class`, and maybe using it as bitfield wasn't a good idea.
It's interesting to hear what other people think about `enum class` as bitfield, maybe I should change it to simple `enum`?

Reply:
For bitfields I would use constexpressions in namespace. Using enum class for it is strange.

Reply:
> For bitfields I would use constexpressions in namespace.

Yea, it's used a lot in our code base, but I don't know, it doesn't seem very informative when you looking at `int` variable in my opinion.
P.S. Sorry for misleading, **bitfields** is something different, I mean **bitflags**, but i think you get the point.

Reply:
Oh, that's an interesting approach, first I thought about unnamed namespace, but now I get it.

Reply:
Also I would call the namespace "Format". In this case `Format::Pattern` and `Format::SSSSS` would be closer to so-called "literate programming". Code readability and self-documenting in one.